core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java

```diff
@@ -89,6 +95,30 @@ public String upcheck() {
 
   @JsonRpcMethod("Filecoin.WalletList")
   public List<String> filecoinWalletList() {
-    return fcKeyIdentifiers.list(KeyType.SECP256K1);
+    return fcSigners.availableIdentifiers().parallelStream().collect(Collectors.toList());
+  }
+
+  @JsonRpcMethod("Filecoin.WalletSign")
+  public Bytes filecoinWalletSign(final String filecoinAddress, final String dataToSign) {
```

Comment:
This is response need to include a type and the signature data so the RPC response becomes something like
```
{"jsonrpc":"2.0","result":{"Type":2,"Data":"p/LeJS8KPzuOco2qpqjhvJxFrA5qj++LpeVoActWkDRQTLhuGsnwAvY1kluQ6PntAqISjvR/RU0kzgkR38F8Hm5NuNPsu9/BEiU+IQheNeD7OoG8THq4OuZMbISg7uR/"},"id":3}
```

Reply:
done